File: backend/app/schemas/purchase.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel
from decimal import Decimal


class PurchaseOrderItemCreate(BaseModel):
    product_id: int
    quantity: int
    unit_price: Optional[Decimal] = None
    is_gift: bool = False


class PurchaseOrderItemResponse(BaseModel):
    id: int
    order_id: int
    product_id: int
    product_name: Optional[str] = None
    product_spec: Optional[str] = None
    product_unit: Optional[str] = None
    quantity: int
    unit_price: Decimal
    subtotal: Decimal
    is_gift: bool

    model_config = {"from_attributes": True}


class PurchaseOrderCreate(BaseModel):
    warehouse_id: int
    items: List[PurchaseOrderItemCreate]
    supplier_name: Optional[str] = None
    supplier_phone: Optional[str] = None
    remark: Optional[str] = None


class PurchaseOrderResponse(BaseModel):
    id: int
    order_no: str
    supplier_name: Optional[str] = None
    supplier_phone: Optional[str] = None
    warehouse_id: int
    warehouse_name: Optional[str] = None
    total_amount: Decimal
    total_quantity: int
    gift_quantity: int
    status: str
    remark: Optional[str] = None
    created_at: datetime
    items: List[PurchaseOrderItemResponse] = []

    model_config = {"from_attributes": True}

File: frontend/src/types/purchase.ts
export interface PurchaseOrderItem {
  id: number;
  order_id: number;
  product_id: number;
  product_name?: string;
  product_spec?: string;
  product_unit?: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
  is_gift: boolean;
}

export interface PurchaseOrderItemCreate {
  product_id: number;
  quantity: number;
  unit_price?: number;
  is_gift: boolean;
}

export interface PurchaseOrder {
  id: number;
  order_no: string;
  supplier_name?: string;
  supplier_phone?: string;
  warehouse_id: number;
  warehouse_name?: string;
  total_amount: number;
  total_quantity: number;
  gift_quantity: number;
  status: string;
  remark?: string;
  created_at: string;
  items: PurchaseOrderItem[];
}

export interface PurchaseOrderCreate {
  warehouse_id: number;
  items: PurchaseOrderItemCreate[];
  supplier_name?: string;
  supplier_phone?: string;
  remark?: string;
}
